site/profile/manifests/core/icinga_master.pp

```diff
@@ -0,0 +1,411 @@
+# @summary
+#   Definition of icinga and icingaweb master module
+
+class profile::core::icinga_master (
+  String $ldap_server,
+  String $ldap_root,
+  String $ldap_user,
+  String $ldap_pwd,
+  String $ldap_resource,
+  String $ldap_user_filter,
+  String $ldap_group_filter,
+  String $ldap_group_base,
+  String $mysql_root,
+  String $mysql_icingaweb_db,
+  String $mysql_icingaweb_user,
+  String $mysql_icingaweb_pwd,
+  String $mysql_director_db,
+  String $mysql_director_user,
+  String $mysql_director_pwd,
+  String $api_name,
+  String $api_user,
+  String $api_pwd,
+  String $credentials_hash,
+){
+  include profile::core::letsencrypt
+  include remi
+  include nginx
+
+#<-------------Variables Definition---------------->
+
+#Implicit usage of facts
+  $master_fqdn  = $facts[fqdn]
+  $master_ip  = $facts[ipaddress]
+
+#Letsencrypt cert path
+  $le_root = "/etc/letsencrypt/live/${master_fqdn}"
+
+#IcingaDirector force Deploy
+  $url         = "https://${master_fqdn}/director"
+  $credentials = "Authorization:Basic ${credentials_hash}"
+  $format      = 'Accept: application/json'
+  $curl        = 'curl -s -k -H'
+  $icinga_path = '/opt/icinga'
+  $deploy_cmd  = "${curl} '${credentials}' -H '${format}' -X POST '${url}/config/deploy'"
+
+#pnp4nagios webpage integration
+  $pnp4nagios_conf = '
+location /pnp4nagios {
+        alias  /usr/share/nagios/html/pnp4nagios;
+        index     index.php;
+        try_files $uri $uri/ @pnp4nagios;
+  }
+location @pnp4nagios {
+        if ( $uri !~ /pnp4nagios/index.php(.*)) {
+                rewrite ^/pnp4nagios/(.*)$ /pnp4nagios/index.php/$1;
+                break;
+        }
+        root /usr/share/nagios/html/pnp4nagios;
+        include /etc/nginx/fastcgi.conf;
+        fastcgi_split_path_info ^(.+\.php)(.*)$;
+        fastcgi_param PATH_INFO $fastcgi_path_info;
+        fastcgi_param SCRIPT_FILENAME $document_root/index.php;
+        fastcgi_pass 127.0.0.1:9000;
+}
+'
+
+#PNP plugin configuration
+  $pnp_conf = '[pnp4nagios]
+config_dir = "/etc/pnp4nagios"
+base_url = "/pnp4nagios"
+menu_disabled = "0"
+default_query = "host=icinga-master.ls.lsst.org&srv=MasterPingService"
+'
+#Icinga tls keys
+  $ssl_cert       = '/etc/ssl/certs/icinga.crt'
+  $ssl_key        = '/etc/ssl/certs/icinga.key'
+
+#Force installation and usage of php73
+  $packages = [
+    'git',
+    'pnp4nagios',
+    'centos-release-scl',
+    'php73-php-fpm',
+    'php73-php-ldap',
+    'php73-php-intl',
+    'php73-php-dom',
+    'php73-php-gd',
+    'php73-php-imagick',
+    'php73-php-mysqlnd',
+    'php73-php-pgsql',
+    'php73-php-pdo',
+    'php73-php-process',
+    'php73-php-cli',
+    'php73-php-soap',
+    'rh-php73-php-posix',
+    'nagios-plugins-all',
+  ]
```

Comment:
I have used https://github.com/voxpupuli/puppet-php to setup fpm for multiple applications in the past including icingaweb. It is always preferrable to use a module that will be maintained over time. I highly encourage you to use that mod. If it doesn't work then php should be split into its own profile.

Reply:
done as suggested

Reply:
What did you do? I do not see a php mod included in the `Puppetfile` and I don't see a new profile.